cpp/cmake_modules/SetupCxxFlags.cmake

```
@@ -329,7 +329,8 @@ if("${BUILD_WARNING_LEVEL}" STREQUAL "CHECKIN")
     set(CXX_COMMON_FLAGS "${CXX_COMMON_FLAGS} -Wno-sign-conversion")
     set(CXX_COMMON_FLAGS "${CXX_COMMON_FLAGS} -Wunused-result")
     set(CXX_COMMON_FLAGS "${CXX_COMMON_FLAGS} -Wdate-time")
-  elseif(CMAKE_CXX_COMPILER_ID STREQUAL "Intel")
+  elseif(CMAKE_CXX_COMPILER_ID STREQUAL "Intel" OR CMAKE_CXX_COMPILER_ID STREQUAL
+                                                   "IntelLLVM")
```

The code further up in the diff is using a regex-match already; would make this a bit easier I guess...
```suggestion
  elseif(CMAKE_CXX_COMPILER_ID Matches "^Intel(LLVM)?$")
```

---

I hope you don't mind, but I'm going to merge the PR as I need it to cherry-pick into an R package release I'm doing today, but we could make this change in a follow-up PR.